### PR TITLE
GROUPAROO_RUN_MODE Bugs

### PR DESCRIPTION
**Pull request by evantahler**
This PR fixes a number of problems on the Grouparoo CLI introduced by https://github.com/grouparoo/grouparoo/pull/2792.  The setting of `process.env.GROPUAROO_RUN_MODE` is atypical in that it is set by the running process itself (part of the init phase of a CLI command).  As such, it is likely that some of other file/module has already loaded the config object and has a local copy in memory.... leading to incorrect values for `config.general.runMode`.  So, this value is not a good candidate for storing in the config.   

This PR reverts those changes, and moves to using a getter utility instead. 
 
### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: core/src/modules/runMode.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const getGrouparooRunMode = () =>
+  process.env.GROUPAROO_RUN_MODE as
+    | "cli:run"
+    | "cli:start"
+    | "cli:config"
+    | "cli:apply"
+    | "cli:validate";
```
Comment (evantahler) [inline core/src/modules/runMode.ts:7]: At least we can still get types

**Review by pedroslopez — COMMENTED** [review]
Bummer :/ 

From #2792, I know we added a bunch of `rebuildConfig()`s to the tests... is this PR missing their removal? I don't think it would be needed anymore.

**Comment by evantahler** [comment]
> From #2792, I know we added a bunch of `rebuildConfig()`s to the tests... is this PR missing their removal? I don't think it would be needed anymore.

Oh yeah!  all of those are removed now via https://github.com/grouparoo/grouparoo/pull/2805/commits/54a9e00f25045cbdf3ca4db158c55abcd93fc1b8

**Review by pedroslopez — APPROVED** [review]
Once tests pass, looks good!

**Pull request merged**